packages/server/lib/controllers/auth/postUnauthenticated.ts

```diff
@@ -13,7 +13,8 @@ import { connectionCreated, connectionCreationFailed } from '../../hooks/hooks.j
 const queryStringValidation = z
     .object({
         connection_id: connectionIdSchema.optional(),
-        public_key: z.string().uuid(),
+        public_key: z.string().uuid().optional(),
+        connect_session_token: z.string().startsWith('nango_connect_session_').optional(),
```

Comment:
note you can use `packages/server/lib/helpers/validation.ts` to store reusable zod schema